release_notes/ocp_3_4_release_notes.adoc

```diff
@@ -0,0 +1,218 @@
+and libraries. {product-title} brings the OpenShift PaaS platform to customer
+data centers, enabling organizations to implement a private PaaS that meets
+security, privacy, compliance, and governance requirements.
+
+[[ocp-34-about-this-release]]
+== About This Release
+
+Red Hat {product-title} version 3.4 is now available. This release is based on
+link:https://github.com/openshift/origin/releases/tag/v1.4.0-rc1[OpenShift Origin 1.4]. New features, changes, bug fixes, and known issues that pertain to
+{product-title} 3.4 are included in this topic.
+
+For initial installations, see the
+xref:../install_config/install/planning.adoc#install-config-install-planning[Installing a Cluster] topics in the
+xref:../install_config/index.adoc#install-config-index[Installation and Configuration] documentation.
+
+To upgrade to this release from a previous version, see the xref:../install_config/upgrading/index.adoc#install-config-upgrading-index[Upgrading a Cluster] topics in the xref:../install_config/index.adoc#install-config-index[Installation and Configuration] documentation.
+
+[[ocp-34-new-features-and-enhancements]]
+== New Features and Enhancements
+
+[[ocp-34-enterprise-container-registry]]
+=== Enterprise Container Registry
+
+This release adds the following improvements to the registry and its user
+experience.
+
+[[ocp-34-developer-experience]]
+=== Developer Experience
+
+This release adds the following improvements to the developer workflow when
+developing and testing applications on {product-title}.
+
+[[ocp-34-web-console]]
+=== Web Console
+
+This release adds the following improvements to the web console, including
+updates to existing features, usability overhauls, and a few brand new concepts.
+
+[[ocp-34-networking]]
+=== Networking
+
+This release adds the following improvements to networking components.
+
+[[ocp-34-security]]
+=== Security
+
+This release adds the following improvements to cluster security.
+
+[[ocp-34-cluster-longevity]]
+=== Cluster Longevity
+
+This release adds the following improvements to cluster longevity.
+
+[[ocp-34-framework-services]]
+=== Framework Services
+
+{product-title} provides resource usage metrics and log access to developers based on the Hawkular and Elasticsearch open source projects. This release adds the following improvements to these components.
+
+[[ocp-34-notable-technical-changes]]
+== Notable Technical Changes
+
+{product-title} 3.4 introduces the following notable technical changes.
+
+[[ocp-34-updated-infrastructure-components]]
+*Updated Infrastructure Components*
+
+- Kubernetes has been updated to v1.4.
+- etcd has been updated to 3.1.0-rc.0.
+- {product-title} 3.4 requires Docker 1.12.
+
+[[ocp-34-updated-elasticsearch]]
+*Updated Elasticsearch in EFK Stack*
+
+The latest EFK stack now uses Elasticsearch 2.4 with a common data model. This
+means Fluentd sends logs to Elasticsearch with a new indexing pattern for
```

Comment:
Confirmed with @ewolinetz that this should now be Elasticsearch 2.4 instead of 2.3 (which we originally had in this note per https://github.com/openshift/openshift-docs/pull/3211, but has changed for OCP 3.4 since then).

Also FYI @danmacpherson @vikram-redhat I moved this note to here in the 3.4 release notes, instead of at the tail end of the logging upgrade section because it felt buried and wasn't specific to the upgrade task. Also fixed a rendering issue in this note due to some superfluous `+`s.